test/testdata/fixtures/multi-dc-dse/k8ssandra.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: k8ssandra.io/v1alpha1
+kind: K8ssandraCluster
+metadata:
+  name: test
+spec:
+  cassandra:
+    serverVersion: "6.8.25"
+    serverType: "dse"
+    storageConfig:
+      cassandraDataVolumeClaimSpec:
+        storageClassName: standard
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 5Gi
+    config:
+      jvmOptions:
+        heapSize: 512Mi
+        heapNewGenSize: 256Mi
+        gc: CMS
+    mgmtAPIHeap: 64Mi
+    datacenters:
```

Comment:
I ran into a problem with having multiple DCs in the same namespace. Cleanup won't happen properly. One of the CassandraDatacenters does not get deleted. The problem stems from the fact all 3 DCs share the same superuser secret which gets deleted some where along the way. I think it might be getting deleted by the replicated secret controller, but I need to verify. 

The secret is deleted before the DC which results in this validation error reported by cass-operator:

`1.6632922466632967e+09	ERROR	controllers.CassandraDatacenter	CassandraDatacenter resource is invalid	{"cassandradatacenter": "multi-dc-dse-8rf4pf/dc3", "requestNamespace": "multi-dc-dse-8rf4pf", "requestName": "dc3", "loopID": "b0473739-0c15-4650-9332-6004e74f1786", "error": "Could not load superuser secret for CassandraCluster: multi-dc-dse-8rf4pf/test-superuser"}`

I tried provided my own superuser secret. Interestingly, it too was deleted. That feels like a bug to me. I don't think the operator should be deleting secrets that it doesn't create. This should be addressed separately.

To avoid the cleanup problem, I would deploy the DCs in separate namespaces.

Reply:
It is in fact the secret controller. The K8ssandraCluster controller adds the `replicatedresource.k8ssandra.io/orphan: "true"` annotation to my superuser secret. The secrets controller however, only looks for that annotation on the ReplicatedSecret. An easy work around would be to add the annotation to the ReplicatedSecret. I tested and verified that this works.

Reply:
Using multiple namespaces would probably be a big change due to how our e2e test framework is built.
I've collocated them in a single cluster/namespace because we had a bug in the image, I'll split them again across multiple clusters now.